mesonbuild/compilers/mixins/clike.py

```diff
@@ -100,6 +100,10 @@ def to_native(self, copy: bool = False) -> T.List[str]:
         # Remove system/default include paths added with -isystem
         default_dirs = self.compiler.get_default_include_dirs()
         if default_dirs:
+            real_default_dirs = [os.path.realpath(i) for i in default_dirs]
+
+            def is_default(opt_dir: str) -> bool:
+                return os.path.realpath(opt_dir) in real_default_dirs
```

Comment:
It feels a bit like this function is longer than just running `os.path.realpath(new[i + 1]) in real_default_dirs` etc. in the 3 places internal to this function that it is needed.